joomla/joomla-cms
# [4.0] Make  lighthouse happy

Pull Request for Issue # .

### Summary of Changes
- Defer any template own JS (exception is the core.js -because reasons-)

Defer the loading of Fontawesome in a Progressive enhancement way:
- The fontawesome css tag is rendered with a `rel="lazy-stylesheet"` and once the page is loaded javascript changes it to stylesheet (forcing it to load)
- The css tag is also rendered inside a noscript tag in the header  for browsers with JS disabled


### Testing Instructions
- Check that Cassiopeia is having all the icons in place
- Disable JS in your browser and check that icons are in place

### Actual result BEFORE applying this Pull Request

![Screenshot 2021-01-24 at 11 52 17](https://user-images.githubusercontent.com/3889375/105628025-a25c3c80-5e3a-11eb-8f1b-c08a514f088b.png)


### Expected result AFTER applying this Pull Request

![Screenshot 2021-01-24 at 11 51 05](https://user-images.githubusercontent.com/3889375/105628004-79d44280-5e3a-11eb-80dc-a2c598e86e60.png)


### Documentation Changes Required

No, but hopefully theme creators and 3rd PD could apply the same simple optimisations so the Joomla ecosystem is really competitive and performance is a serious selling point!!!

FWIW Google will start using Core Web Vitals from May 2021: https://developers.google.com/search/blog/2020/11/timing-for-page-experience In short **performance will affect the SEO ranking**
